pkg/roachpb/errors.proto

```diff
@@ -184,8 +184,15 @@ message SendError {
 // An AmbiguousResultError indicates that an EndTransaction request
```

Comment:
I think this entire comment needs to be revised. How about:
```
// An AmbiguousResultError indicates that a write request
// may have succeeded or failed, but the response was not
// received and the final result is ambiguous.
```
The other verbiage you added seems too specific and it's confusing to boot. This error is not just returned for non-transactional batches in the case you mention with `WriteTooOldError`s. It's also returned for all other errors which occur after a Raft re-proposal.